pkg/common/structurev2/standard.go

```diff
@@ -78,3 +197,127 @@ func (n *StandardMapNode) Children() NodeChildrenIterator {
 		}
 	}
 }
+
+// MarshalYAML implements yaml.Marshaler.
+func (n *StandardMapNode) MarshalYAML() (interface{}, error) {
+	mapNode := &yaml.Node{
+		Kind:    yaml.MappingNode,
+		Tag:     "!!map",
+		Content: []*yaml.Node{},
+	}
+	for k, child := range n.Children() {
+		key := k.Key
+		marshaller, ok := child.(yaml.Marshaler)
+		if !ok {
+			return nil, fmt.Errorf("map node child is not implementing yaml.Marshaller")
+		}
+		node, err := marshaller.MarshalYAML()
+		if err != nil {
+			return nil, err
+		}
+		yamlNode, ok := node.(*yaml.Node)
+		if !ok {
+			return nil, fmt.Errorf("map node child returned non *yaml.Node as its result")
+		}
+		keyNode := &yaml.Node{
+			Kind:  yaml.ScalarNode,
+			Value: key,
+			Tag:   "!!str",
+		}
+		mapNode.Content = append(mapNode.Content, keyNode)
+		mapNode.Content = append(mapNode.Content, yamlNode)
+	}
+	return mapNode, nil
+}
+
+// MarshalJSON implements json.Marshaler.
+func (n *StandardMapNode) MarshalJSON() ([]byte, error) {
+	var buf bytes.Buffer
+	buf.WriteString("{")
+	for i, child := range n.Children() {
+		if i.Index > 0 {
+			buf.WriteString(",")
+		}
+		key := fmt.Sprintf("\"%s\"", i.Key)
+		buf.WriteString(key)
+		buf.WriteString(":")
+		marshaller, ok := child.(json.Marshaler)
+		if !ok {
+			return nil, fmt.Errorf("map node child is not implementing json.Marshaller")
+		}
+		marshalled, err := marshaller.MarshalJSON()
+		if err != nil {
+			return nil, err
+		}
+		buf.Write(marshalled)
+	}
+	buf.WriteString("}")
+
+	return buf.Bytes(), nil
+
+}
+
+var _ Node = (*StandardMapNode)(nil)
+var _ yaml.Marshaler = (*StandardMapNode)(nil)
+var _ json.Marshaler = (*StandardMapNode)(nil)
+
+// getYAMLMarshaler returns the yaml.Marshaller from Node interface.
+func getYAMLMarshaler(node Node) (yaml.Marshaler, error) {
+	standardRootNode, err := cloneStandardNodeFromNode(node)
+	if err != nil {
+		return nil, err
+	}
+	return standardRootNode.(yaml.Marshaler), nil
+}
+
+// getJSONMarshalerFromNode returns the json.Marshaller from Node interface.
+func getJSONMarshalerFromNode(node Node) (json.Marshaler, error) {
+	standardRootNode, err := cloneStandardNodeFromNode(node)
+	if err != nil {
+		return nil, err
+	}
+	return standardRootNode.(json.Marshaler), nil
+}
+
+// cloneStandardNodeFromNode clones Node interface into Standard***Node.
+// Standard**Node implements json.Marshaller and yaml.Marshaller, it allows serializing any implementing Node interface.
+func cloneStandardNodeFromNode(node Node) (Node, error) {
```

Comment:
I believe this part is the most important to handle large objects, but I don't see any issue so far.